### PR TITLE
Update DJLModel class for latest container releases

### PR DESCRIPTION
**Pull request by siddvenk**
*Issue #, if available:*

*Description of changes:*

The DJLModel class has not been updated in multiple container releases, and as such it's functionality is completely broken. 

This change updates the DJLModel class for the latest DJL container releases. Furthermore, it simplifies the interface and functionality so that it is more future-proofed. 

Specifically:
- Removes the subclasses of DJLModel. All functionality and auto-configuration behavior is handled partly by the DJLModel class, but mostly by the container at runtime
- Removes the configuration file `serving.properties` requirement in favor of environment variables. A serving.properties file can still be provided and used, but the DJLModel class will not read/parse this configuration.
- Updates the interactions with ModelBuilder interface to reflect the changes here

*Testing done:*
- Updated unit tests
- Used DJLModel to deploy SM endpoints in dev account

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/frameworks/djl/using_djl.rst [hunk 1]
```diff
@@ -19,238 +16,52 @@ For general information about using the SageMaker Python SDK, see :ref:`overview
...
+    djl_model = DJLModel(
+        model_id="<hf hub id | s3 uri>",
+        role="my_sagemaker_role",
         task="text-generation",
-        tensor_parallel_degree=2, # number of gpus to partition the model across
+        engine="Python",
+        env={
+            "OPTION_ROLLING_BATCH": "lmi-dist",
+            "OPTION_DTYPE": "bf16",
+            "MAX_BATCH_SIZE": "256",
```
Comment (frankfliu) [inline doc/frameworks/djl/using_djl.rst:47]:
```suggestion
            "OPTION_MAX_ROLLING_BATCH_SIZE": "64",
```

File: doc/frameworks/djl/using_djl.rst [hunk 2]
```diff
@@ -19,238 +16,52 @@ For general information about using the SageMaker Python SDK, see :ref:`overview
...
-        "my_sagemaker_role",
-        data_type="fp16",
+    djl_model = DJLModel(
+        model_id="<hf hub id | s3 uri>",
+        role="my_sagemaker_role",
         task="text-generation",
-        tensor_parallel_degree=2, # number of gpus to partition the model across
+        engine="Python",
+        env={
+            "OPTION_ROLLING_BATCH": "lmi-dist",
```
Comment (frankfliu) [inline doc/frameworks/djl/using_djl.rst:45]: Maybe add 

"TENSOR_PARALLEL_DEGREE": "2",

File: doc/frameworks/djl/using_djl.rst (hunk 1 above)
Comment (siddvenk) [reply doc/frameworks/djl/using_djl.rst]: done

File: doc/frameworks/djl/using_djl.rst (hunk 2 above)
Comment (siddvenk) [reply doc/frameworks/djl/using_djl.rst:45]: done

**Review by frankfliu — APPROVED** [review]

File: doc/frameworks/djl/using_djl.rst [hunk 3]
```diff
@@ -19,238 +16,53 @@ For general information about using the SageMaker Python SDK, see :ref:`overview
...
 
-    # Deploy the model to an Amazon SageMaker Endpoint and get a Predictor
-    deepspeed_predictor = deepspeed_model.deploy("ml.g5.12xlarge",
-                                                 initial_instance_count=1)
-    hf_accelerate_predictor = hf_accelerate_model.deploy("ml.g5.12xlarge",
-                                                         initial_instance_count=1)
-    fastertransformer_predictor = fastertransformer_model.deploy("ml.g5.12xlarge",
-                                                                 initial_instance_count=1)
-
-Regardless of which way you choose to create your model, a ``Predictor`` object is returned. You can use this ``Predictor``
```
Comment (grenmester) [inline doc/frameworks/djl/using_djl.rst:84]: should we still leave in some docs explaining that a `Predictor` is returned on `deploy()`?

File: doc/frameworks/djl/using_djl.rst [hunk 4]
```diff
@@ -19,238 +16,53 @@ For general information about using the SageMaker Python SDK, see :ref:`overview
...
-to do inference on the endpoint hosting your DJLModel.
-
 Each ``Predictor`` provides a ``predict`` method, which can do inference with json data, numpy arrays, or Python lists.
 Inference data are serialized and sent to the DJL Serving model server by an ``InvokeEndpoint`` SageMaker operation. The
 ``predict`` method returns the result of inference against your model.
 
 By default, the inference data is serialized to a json string, and the inference result is a Python dictionary.
 
-Model Directory Structure
```
Comment (grenmester) [inline doc/frameworks/djl/using_djl.rst:93]: is all of this removed information being moved to the LMI docs on the AWS documentation site?

File: src/sagemaker/serve/utils/hf_utils.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import urllib.request
+from json import JSONDecodeError
+from urllib.error import HTTPError, URLError
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def _get_model_config_properties_from_hf(model_id: str, hf_hub_token: str = None):
+    """Placeholder docstring"""
```
Comment (grenmester) [inline src/sagemaker/serve/utils/hf_utils.py:25]: docstring

File: src/sagemaker/serve/model_server/djl_serving/utils.py [hunk 6]
```diff
@@ -144,22 +96,23 @@ def _get_default_max_tokens(sample_input, sample_output) -> tuple:
     return (max_total_tokens, max_new_tokens)
 
 
-def _set_serve_properties(hf_model_config: dict, schema_builder: SchemaBuilder) -> tuple:
+def _get_default_djl_configurations(
+    model_id: str, hf_model_config: dict, schema_builder: SchemaBuilder
+) -> tuple:
```
Comment (grenmester) [inline src/sagemaker/serve/model_server/djl_serving/utils.py:101]: nit: tuple[dict, int]

File: src/sagemaker/djl_inference/model.py [hunk 7]
```diff
@@ -354,66 +97,113 @@ def __init__(
...
+                "both model_id and model_data are provided. Please only provide one of them"
+            )
+
+    def _infer_engine(self) -> Optional[str]:
+        """Placeholder docstring"""
+        if self.engine is not None:
+            logger.info("Using provided engine %s", self.engine)
+            return self.engine
+
+        if self.task is not None:
```
Comment (grenmester) [inline src/sagemaker/djl_inference/model.py:150]: nit: unnecessary, `None == "text-embedding"` is not an error

**Review by grenmester — APPROVED** [review]

File: doc/frameworks/djl/using_djl.rst (hunk 4 above)
Comment (siddvenk) [reply doc/frameworks/djl/using_djl.rst:93]: Yes, we have this all on our LMI docs site.

We are not publishing on AWS docs anymore, but have aligned on all our docs living here https://docs.djl.ai/docs/serving/serving/docs/lmi/index.html. The model directory structure specifically is here https://docs.djl.ai/docs/serving/serving/docs/lmi/deployment_guide/model-artifacts.html

File: doc/frameworks/djl/using_djl.rst (hunk 3 above)
Comment (siddvenk) [reply doc/frameworks/djl/using_djl.rst:84]: yea i can do that, I think i deleted this line by mistake!

**Comment by siddvenk** [comment]
@akrishna1995 - can you review this PR? I have two approvals, one from my team (owning DJLModel class), and one from the team that owns the ModelBuilder class.

**Review by mufaddal-rohawala — APPROVED** [review]

**Pull request merged**